### PR TITLE
fix: check traversal_paths once

### PR DESCRIPTION
**Pull request by alaeddine-13**
closes https://github.com/jina-ai/jina/issues/3947

### CODE REVIEW

File: jina/types/arrays/mixins/traverse.py [hunk 1]
```diff
@@ -48,13 +48,15 @@ class TraverseMixin:
     def traverse(
         self: 'T',
         traversal_paths: str,
+        check_traversal_paths: bool = True,
```
Comment (JoanFM) [inline jina/types/arrays/mixins/traverse.py:51]: Why is this needed? I do not like adding this to the public interface, this should not be cared by the user

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/types/arrays/mixins/traverse.py [hunk 2]
```diff
@@ -18,16 +18,13 @@
...
         is_deprecate_scheme = False
-
-    if not (
-        tp
-        and not isinstance(tp, str)
-        and isinstance(tp, Sequence)
-        and all(isinstance(p, str) for p in tp)
-    ):
-        raise TypeError('`traversal_paths` needs to be `Sequence[str]`')
+    elif isinstance(tp, Sequence) and all(isinstance(p, str) for p in tp):
```
Comment (JoanFM) [inline jina/types/arrays/mixins/traverse.py:22]: no need to check `is_deprecate_warning`. Just raise the `warning` here and that's it

**Review by JoanFM — CHANGES_REQUESTED** [review]
Looks good, just a minor thing

File: jina/types/arrays/mixins/traverse.py (hunk 2 above)
Comment (alaeddine-13) [reply jina/types/arrays/mixins/traverse.py:21]: changed

**Review by hanxiao — APPROVED** [review]
